Source/Scene/Scene.js

```diff
@@ -868,11 +868,9 @@ define([
                 break;
             }
 
-            if (command.pass === Pass.OPAQUE || command instanceof ClearCommand) {
-                frustumCommands.opaqueCommands[frustumCommands.opaqueIndex++] = command;
-            } else if (command.pass === Pass.TRANSLUCENT){
-                frustumCommands.translucentCommands[frustumCommands.translucentIndex++] = command;
-            }
+            var pass = command instanceof ClearCommand ? Pass.OPAQUE : command.pass;
+            var index = frustumCommands.indices[pass]++;
+            frustumCommands.commands[pass][index] = command;
```

Comment:
Right, I think we talked about this before. We need some heuristic for estimating how much/when to grow/shrink the array. We did it this way because the number of commands won't change too much frame to frame. The only time the array would grow rapidly is the first time you zoom way in and the number of tiles rendered increases. The same goes when adding a bunch of primitives.